Failing unit test in webhook validation after dependency bump (grpc 1.79.3)

The Pipelines as Code CI is failing on `main` after the recent dependency update (`google.golang.org/grpc` from `1.78.0` to `1.79.3`). The failure occurs in the `unit` test task, specifically within webhook validation tests.

**Failure details**
PipelineRun: `push-generate-coverage-releaseyaml-647q9`
Namespace: `pipelines-as-code-ci`

```
FAIL github.com/openshift-pipelines/pipelines-as-code/pkg/webhook 1.151s
make: *** [Makefile:70: test-unit] Error 1
```

**Error location**

```
validation_test.go:195
assertion failed: true (res.Allowed bool) != false (tt.allowed bool)
```

**Observed behavior**
A test expectation in `validation_test.go` no longer matches the actual result. The test expects `Allowed=false`, but the code returns `Allowed=true`.

**Expected behavior**
Unit tests should pass consistently on `main`. The validation logic should either:

* remain unchanged and tests should reflect correct expectations, or
* tests should be updated if the behavior change is intentional.

**Context**
This started after:

```
chore(deps): bump google.golang.org/grpc from 1.78.0 to 1.79.3
```

**Potential causes**

* Behavioral change introduced by grpc update
* Side effect in webhook validation logic (possibly admission handling or context propagation)
* Test relying on assumptions affected by dependency behavior

**Suggested actions**

* Identify what changed in validation logic or grpc interaction
* Confirm whether the new behavior is correct
* Update test expectations if intentional, otherwise fix regression

**Additional info**
CI link indicates failure in:
`Pipelines as Code Dogfooding CI / push-generate-coverage-releaseyaml`

--- a/pkg/webhook/validation.go
+++ b/pkg/webhook/validation.go
@@ -46,7 +46,11 @@ func (ac *reconciler) Admit(_ context.Context, request *v1.AdmissionRequest) *v1
 			return webhook.MakeErrorStatus("URL must be set")
 		}
 
-		if err := validateRepositoryURL(repo.Spec.URL); err != nil {
+		var gitProviderType string
+		if repo.Spec.GitProvider != nil {
+			gitProviderType = repo.Spec.GitProvider.Type
+		}
+		if err := validateRepositoryURL(repo.Spec.URL, gitProviderType); err != nil {
 			return webhook.MakeErrorStatus("%s", err.Error())
 		}
 	}
@@ -94,7 +98,7 @@ func checkIfRepoExist(pac pac.RepositoryLister, repo *v1alpha1.Repository, ns st
 	return false, nil
 }
 
-func validateRepositoryURL(repoURL string) error {
+func validateRepositoryURL(repoURL, gitProviderType string) error {
 	parsedURL, err := url.Parse(repoURL)
 	if err != nil {
 		return fmt.Errorf("invalid URL format: %w", err)
@@ -108,7 +112,7 @@ func validateRepositoryURL(repoURL string) error {
 	// (like https://github.com/org/repo/extra).
 	// Detect if this is a GitHub instance (github.com or GHE) by checking headers
 	//  and API endpoints.
-	if isGitHubInstance(parsedURL.Host, parsedURL.Scheme) {
+	if isGitHubInstance(parsedURL.Host, parsedURL.Scheme, gitProviderType) {
 		// Remove leading and trailing "/"
 		repoPath := strings.Trim(parsedURL.Path, "/")
 
@@ -122,8 +126,12 @@ func validateRepositoryURL(repoURL string) error {
 }
 
 // isGitHubInstance detects if a host is github.com or a GitHub Enterprise instance.
-// It checks the server header and /api/v3/meta endpoint.
-func isGitHubInstance(host, scheme string) bool {
+// It checks the provider type first, then the host, and falls back to HTTP detection.
+func isGitHubInstance(host, scheme, gitProviderType string) bool {
+	if gitProviderType == "github" {
+		return true
+	}
+
 	if host == "github.com" {
 		return true
 	}

--- a/pkg/webhook/validation_test.go
+++ b/pkg/webhook/validation_test.go
@@ -128,6 +128,7 @@ func TestReconciler_Admit(t *testing.T) {
 				Name:             "test-run",
 				InstallNamespace: "namespace",
 				URL:              "https://ghe.pipelinesascode.com/owner/repo/subgroup",
+				GitProviderType:  "github",
 			}),
 			allowed: false,
 			result:  "github repository URL must follow https://github.com/org/repo format without subgroups (found 3 path segments, expected 2): https://ghe.pipelinesascode.com/owner/repo/subgroup",

